docs/src/tutorials/conic/optimal-power-flow.jl

```diff
@@ -0,0 +1,286 @@
+# Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors    #src
+# This Source Code Form is subject to the terms of the Mozilla Public License   #src
+# v.2.0. If a copy of the MPL was not distributed with this file, You can       #src
+# obtain one at https://mozilla.org/MPL/2.0/.                                   #src
+
+# # Optimal power flow
+
+# **This tutorial was originally contributed by James Foster (@jd-foster).**
+
+# This tutorial formulates and solves an optimal power flow problem,
+# a much-studied nonlinear problem from the field of electrical engineering.
+# Our particular focus is on obtaining a good estimate of the objective value
+# at the global optimum through the use of semidefinite programming techniques.
+
+# The tutorial highlights JuMP's ability to formulate problems involving
+# complex-valued decision variables and complex matrix cones such as the 
+# [`HermitianPSDCone`](@ref) object.
+# See the [Complex number support](@ref) section of the manual for more details.
+
+# For another example of modeling with complex decision variables
+# see the [Quantum state discrimination](@ref) tutorial.
+
+# ## Required packages
+
+# This tutorial requires the following packages:
+
+using JuMP
+import Ipopt
+import SCS
+import LinearAlgebra: diag, diagm
+import SparseArrays: sparse, spdiagm, sparsevec
+import DataFrames: DataFrame, rename
```

Comment:
I think I prefer `LinearAlgebra.diag` etc in the text. It's more verbose, but it makes it much clearer where things are coming from.

Reply:
Ok, I've done the `LinearAlgebra` ones; did you mean for the other packages too?

Reply:
Yeah